smithy-rules-engine/src/main/java/software/amazon/smithy/rulesengine/traits/ContextParamDefinition.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.smithy.rulesengine.traits;
+
+import java.util.Objects;
+import software.amazon.smithy.utils.SmithyBuilder;
+import software.amazon.smithy.utils.SmithyUnstableApi;
+
+@SmithyUnstableApi
+public final class ContextParamDefinition {
```

Comment:
should implement ToSmithyBuilder

Reply:
Why make this a class (and the other Definition classes)? Why not just use ContextParamTrait and provide access to `name` through there instead of adding this layer? We typically use the traits themselves to carry their own information.